src/client/eppo-client-assignment-details.spec.ts

```diff
@@ -58,38 +63,40 @@ describe('EppoClient get*AssignmentDetails', () => {
     );
     const expected: IAssignmentDetails<number> = {
       value: 3,
-      environmentName: 'Test',
-      variationKey: 'three',
-      variationValue: 3,
-      flagEvaluationCode: 'MATCH',
-      flagEvaluationDescription:
-        'Supplied attributes match rules defined in allocation "targeted allocation".',
-      configFetchedAt: expect.any(String),
-      configPublishedAt: expect.any(String),
-      matchedRule: {
-        conditions: [
+      evaluationDetails: {
+        environmentName: 'Test',
+        variationKey: 'three',
+        variationValue: 3,
+        flagEvaluationCode: 'MATCH',
+        flagEvaluationDescription:
+          'Supplied attributes match rules defined in allocation "targeted allocation".',
+        configFetchedAt: expect.any(String),
+        configPublishedAt: expect.any(String),
+        matchedRule: {
+          conditions: [
+            {
+              attribute: 'country',
+              operator: OperatorType.ONE_OF,
+              value: ['US', 'Canada', 'Mexico'],
+            },
+          ],
+        },
+        matchedAllocation: {
+          key: 'targeted allocation',
+          allocationEvaluationCode: AllocationEvaluationCode.MATCH,
+          orderPosition: 1,
+        },
+        unmatchedAllocations: [],
+        unevaluatedAllocations: [
           {
-            attribute: 'country',
-            operator: OperatorType.ONE_OF,
-            value: ['US', 'Canada', 'Mexico'],
+            key: '50/50 split',
+            allocationEvaluationCode: AllocationEvaluationCode.UNEVALUATED,
+            orderPosition: 2,
           },
         ],
       },
-      matchedAllocation: {
-        key: 'targeted allocation',
-        allocationEvaluationCode: AllocationEvaluationCode.MATCH,
-        orderPosition: 1,
-      },
-      unmatchedAllocations: [],
-      unevaluatedAllocations: [
-        {
-          key: '50/50 split',
-          allocationEvaluationCode: AllocationEvaluationCode.UNEVALUATED,
-          orderPosition: 2,
-        },
-      ],
     };
-    expect(result).toMatchObject(expected);
+    expect(result).toEqual(expected);
```

Comment:
cool to know `toEqual()` still plays nicely with `expect.any(String)`

Note that given the size of the object this will still be a bit more annoying than an expect per field, but not too annoying.

You can see a (LARGE 😅 ) example of what I prefer [here](https://github.com/Eppo-exp/eppo/blob/main/packages/api/src/randomized-assignment/randomized-assignment.controller.e2e-spec.ts#L614).